R/convertFromBIOM.R

```diff
@@ -274,16 +286,9 @@ convertFromBIOM <- function(
     # Get column
     col = x[ , colname]
     # List prefixes
-    prefixes <- c(
-        "^d__",
-        "^k__",
-        "^p__",
-        "^c__",
-        "^o__",
-        "^f__",
-        "^g__",
-        "^s__"
-    )
+    all_ranks <- .get_all_supported_ranks()
+    prefixes <- paste0("^", all_ranks, "__")
+    names(prefixes) <- names(all_ranks)
```

Comment:
Here also: taxonomy ranks are detected based on prefixes of all supported ranks in mia.